src/test/java/com/solvd/laba/carina/demo/MyFitnessPalTest.java

```diff
@@ -0,0 +1,137 @@
+    public void testSignUp(Account account){
+        HomePage homepage = new HomePage(getDriver());
+        LOGGER.trace("Opening https://www.myfitnesspal.com");
+        homepage.open();
+        LOGGER.trace("Assert page is opened");
+        homepage.assertPageOpened();
+
+        LOGGER.trace("Attempt to click Accept cookies button");
+        if (homepage.checkCookiesModal()) {
+            homepage.clickAcceptCookiesButton();
+        }
+        LOGGER.trace("Assert that cookies modal has been dismissed");
+        Assert.assertFalse(homepage.checkCookiesModal(), "Cookies message could not be dismissed.");
+
+        LOGGER.trace("Attempt to click sign up button.");
+        CreateAccountPage createAccountPage = homepage.clickSignUpButton();
+
+        LOGGER.trace("Attempt to click Continue button.");
+        createAccountPage.clickContinue();
+
+        LOGGER.trace("Attempt to submit first name");
+        createAccountPage.enterName(account);
+        LOGGER.trace("Assert that first name page step has been completed");
+        Assert.assertFalse(createAccountPage.isEnterNamePage(), "Enter name page could not be completed correctly.");
+
+        LOGGER.trace("Attempt to submit goals");
+        for (Goal goal : account.getGoals()) {
+            createAccountPage.enterGoals(goal);
+            Assert.assertTrue(createAccountPage.isGoalClicked(goal));
+        }
+
+        if (account.getGoals().size() < 3){
+            createAccountPage.continueFromGoals();
+        }
+
+        LOGGER.trace("Assert that goal page step has been completed");
+        Assert.assertFalse(createAccountPage.isGoalsPage(), "Goal page could not be completed correctly.");
+
+        LOGGER.trace("Attempts to continue from BigStep page");
+        createAccountPage.continueFromBigStep();
+        LOGGER.trace("Assert that goal page step has been completed");
+        Assert.assertFalse(createAccountPage.isBigStepPage(), "BigStep page could not be completed correctly.");
+
+
+        List<Goal> orderedGoals = account.getGoals().stream().sorted(Comparator.naturalOrder()).collect(Collectors.toList());
+        for (Goal goal : account.getGoals()) {
+            switch(goal){
+                case LOSE_WEIGHT: processWeightLossGoal(account, createAccountPage);
+                case MAINTAIN_WEIGHT: processWeightMaintainGoal(account, createAccountPage);
+                case GAIN_WEIGHT: processWeightGainGoal(account, createAccountPage);
+                case GAIN_MUSCLE:processGainMuscleGoal(account, createAccountPage);
+                case MODIFY_DIET:processModifyDietGoal(account, createAccountPage);
+                case MANAGE_STRESS:processManageStressGoal(account, createAccountPage);
+                case INCREASE_STEP_COUNT:processIncreaseStepCount(account, createAccountPage);
+            }
+        }
+    }
+
+    public void processWeightLossGoal(Account account, CreateAccountPage createAccountPage){
```

Comment:
This can be a generic method on the CreateAccountPage. You can get goals from account and get an enum with the name of the screen (you need to create it)